### ISSUE TITLE
3.1 - Consider blank menu icons

### ISSUE THREAD

**Issue by tresf** [dorkbox/SystemTray#42]
On all major Ubuntu versions, when a menu is constructed with an icon, it shifts the text to the right.
For items without menu icons, they appear shifted to the left.

In the event icons are used, I would like to propose the consideration of re-building the menu with blank icons for items without, for better alignment (or an API method to do this).  ElementaryOS seems to [do this too](https://cloud.githubusercontent.com/assets/10621658/21517104/1a0429d6-ccaa-11e6-9f97-cfecfc70c391.png).

The menu items "Automatically start", "Block anonymous requests" and "Show all notifications" suffer this problem.

If this is a technical limitation, please close as won't-fix.

**Ubuntu 12.04**
> ![image](https://cloud.githubusercontent.com/assets/6345473/22052848/e61df94a-dd1a-11e6-9238-f9793c0793fb.png)

**Ubuntu 14.04:**

> ![image](https://cloud.githubusercontent.com/assets/6345473/22052709/bd78b58a-dd19-11e6-990d-ac2c101eefe6.png)

**Ubuntu 16.04:**

> ![image](https://cloud.githubusercontent.com/assets/6345473/22052063/b7082d1a-dd15-11e6-9229-fed8f53d0816.png)

**Comment by dorkbox** [comment]
Since there are enough changes to 3.0 as it is, I'd like to move this to 3.1.

**Comment by tresf** [comment]
@dorkbox understood.  We can do platform detection and attempt to fix this downstream by proving a blank image for now.

FYI, on the right side of GitHub, there's a milestone area that can be used instead of prefixing the titles if interested.  The same milestone can be used as a bug report filter when browsing the issues listing.

**Comment by dorkbox** [comment]
@tresf, I looked into this in more detail -- it only appears to happen with AppIndicator menus (GtkStatusIcon menu's are fine), even though they are both the same GTK menus. More specifically, I only verified this this on Ubuntu (and ubuntu-based distros) -- I don't know if this is happening on Fedora, etc.

Unfortunately, Checkbox items cannot accept an image -- I can change it to an image of a checkbox (this is what I do on Swing, because of the same spacing-problem), and simulate checkbox behavior as a "normal" menu entry, however the color is wrong as it is always a black checkbox. 

On Ubuntu, the default theme is white/gray text on a dark background, however that can be changed with a theme -- and is different on other distros. This will likely be a 3.1 fix, simply because there is a lot of extra checking: i have to be able to query/determine the font color for the menu entry..., then I can change the checkbox image to the appropriate color.
